examples/systemcalls/systemcalls.c

```diff
@@ -92,6 +128,41 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
+    int status;
+    pid_t pid;
+    int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
+    if (fd < 0)
+    {
+        perror("open");
+    }
+    pid = fork();
+    if (pid == -1)
+    {
+        return false;
+    }
+    else if (pid == 0) 
+    {
+        dup2(fd, 1);
+        execv(command[0],command);
+        exit (-1);
+    }
+    if(waitpid(pid, &status, 0) == -1)
+    {
+        return false;
+    }
+    else if (WIFEXITED (status))
+    {
+        int exit_status = WEXITSTATUS(status);
+        if (exit_status == status)
+        {
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+    
```

Comment:
@TejaswiniLakshminarayan think about how you could avoid duplicating logic between the two do_exec functions.